crates/rolldown_plugin/src/plugin_driver/build_hooks.rs

```diff
@@ -238,17 +237,23 @@ impl PluginDriver {
       if map.get_source_content(0).map_or(true, str::is_empty) {
         map.set_source_contents(vec![original_code]);
       }
-      map
+      Some(map)
+    } else if let Some(code) = code {
+      if original_code == code {
```

Comment:
Seems a high cost operation if code doesn't change.

Does rollup have this too?

Reply:
I'll suggest if user explicitly return the code, we consider it changed no matter if really changes or not.

Reply:
> Seems a high cost operation if code doesn't change.
> 
> Does rollup have this too?

The rollup sourcemap implemention is different with rolldown.



> I'll suggest if user explicitly return the code, we consider it changed no matter if really changes or not.

Here look like has a high overhead, but it only happened at only return code and not return map, here we need to create sourcemap for it and also need to remapping, It is worth to using a pre-check to avoid it. 
Yeah. It has overhead for the code actually changed and the code length is not changed. But the fetching module and render module is running at parallel, make the slow task as much as possible faster finish will get better performance.